backend/api/api_key_views.py

```diff
@@ -30,12 +30,14 @@ def api_keys(
     ) -> Response:
         """Custom action to fetch api keys of an api deployment."""
         if api_id:
-            api = DeploymentHelper.get_api_by_id(api_id=api_id)
+            api = DeploymentHelper.get_api_by_id(api_id=api_id, user=request.user)
```

Comment:
@muhammad-ali-e do you think [permission classes](https://github.com/Zipstack/unstract/blob/8d21cb1cd11530554bb8e8b867586a4a7a13f3a6/backend/api/api_deployment_views.py#L103) can help here?

Reply:
@chandrasekharan-zipstack I have some clarifications regaurding this permission class. like it applied on after queriying or before.Let me check it, we can improve it later.

Reply:
@chandrasekharan-zipstack 
After reviewing the code, I believe using the permission class is more appropriate than applying a filter. Since we’re fetching a single entry from the database, applying the filter after retrieval is acceptable and requires minimal changes.

Changed to permission class